src/server.rs

```diff
@@ -889,9 +900,9 @@ impl Server {
 
         let leader_id = u32::from_be_bytes(data[0..4].try_into().unwrap());
         let current_term = u32::from_be_bytes(data[4..8].try_into().unwrap());
-        let commit_index = u32::from_be_bytes(data[8..12].try_into().unwrap());
-        let previous_log_index = u32::from_be_bytes(data[12..16].try_into().unwrap());
-        let peers_count = u32::from_be_bytes(data[16..].try_into().unwrap());
+        let commit_index = u32::from_be_bytes(data[12..16].try_into().unwrap());
+        let previous_log_index = u32::from_be_bytes(data[16..20].try_into().unwrap());
+        let peers_count = u32::from_be_bytes(data[20..24].try_into().unwrap());
```

Comment:
curious, what do we have in data[8..12]?

Reply:
This is because of the following code snippet, where `[8..12]` is the request type for `RPC`:
https://github.com/spacewalkhq/raft-rs/blob/e6b96b0be3f42adc3cce4ed214d35fbb2f2a5539/src/server.rs#L864-L872

Reply:
Thank you.